### PR TITLE
some small changes i have foundt useful

### PR DESCRIPTION
**Pull request by Tkaag**

### CODE REVIEW

**Comment by pvaiko** [comment]
@Tkaag a bit more description about what these changes do would be great so we know what problem are we solving with these.

**Comment by Tkaag** [comment]
The changes to CombineUnloadAIDriver.lua will keep the unload ai alongside the combine like a chopper as is normal for potato and sugarbeet harvest with trailed harvester on big fields. this is done by checking the followAtFillLevel. this update is to stop the unload driver stopping just to start again and chase down the harvester. 

The changes to AIDriver.lua will make a reversing unload try to get back to the same starting point, instead of unloading where the last unload drive left off. because this is a problem with big trailers, where the chaff in a bunker easily reaches from the unloading point the end of the bunker. after just a few full loads. with the changes, the unloading drive will try to get to the staring unloading point every time, and therefore it is better at filling big bunker from one end.
That leads us to the last to changes. Changes to BunkersiloManager.lua is adding an overlap to mode 10 and 9. with the overlap in mode 10 the driver will slowly spread the chaff from the middle of the bunker, where it is unloaded, to the sides of the bunker. and the extra offset is mostly because in my testing the mode 10 derive always get stock om the sides of the bunker, that may be user error. The changes in LevelCompactAIDriver.lua will make the mode 10 driver to the middle of the bunker before driving back. Because when reverse unloading there is not a fill level in the first to columns that the mode 10 driver normally will drive to.

love your mod, and fs will never be the same without.

**Comment by Tensuko** [comment]
That sounds very interesting, but I got one concern about the reverse unload change.
Did you try it with few Trailers, also smal ones ? I Think they can get stuck more easily or 'fly arround' when drive up a hill to unload, or did I get it wrong ?

**Comment by Tkaag** [comment]
I have not testet it with small trailers. That I will do next time I open fs. But the trailer should not get stuck because when the tractor stat slipping it will start unload, and there for the trailer will not get stuck do to it becoming lighter. I will gladly make a video of it.

**Comment by Tensuko** [comment]
That sounds good :)

File: CombineUnloadAIDriver.lua [hunk 1]
```diff
@@ -1805,7 +1805,7 @@ function CombineUnloadAIDriver:unloadMovingCombine()
 	end
 
 	--when the combine is empty, stop and wait for next combine
-	if self:getCombinesFillLevelPercent() <= 0.1 then
+	if self:getCombinesFillLevelPercent() <= 0.1 and self.vehicle.cp.followAtFillLevel >0 then
```
Comment (pvaiko) [inline CombineUnloadAIDriver.lua:1808]: @schwiti6190 refactored the settings in the meanwhile, vehicle.cp.followAtFillLevel is now an object, use vehicle.cp.settings.followAtFillLevel:get() (or self:getFillLevelThreshold()

File: BunkersiloManager.lua [hunk 2]
```diff
@@ -40,7 +40,7 @@ function BunkerSiloManager:createBunkerSiloMap(vehicle, Silo,width, height)
 	local heightLengthZ = (Silo.bunkerSiloArea.hz-Silo.bunkerSiloArea.sz)/heightCount
 	local widthLengthX = (Silo.bunkerSiloArea.wx-Silo.bunkerSiloArea.sx)/widthCount
 	local widthLengthZ = (Silo.bunkerSiloArea.wz-Silo.bunkerSiloArea.sz)/widthCount
-	local getOffTheWall = 0.5;
+	local getOffTheWall = vehicle.cp.workWidth*0.4;--distens from wall is 40% of the working distens
```
Comment (pvaiko) [inline BunkersiloManager.lua:43]: typo

File: BunkersiloManager.lua [hunk 3]
```diff
@@ -26,7 +26,7 @@ function BunkerSiloManager:createBunkerSiloMap(vehicle, Silo,width, height)
 	if 	width then
 		widthCount =math.ceil(bunkerWidth/width)
 	else
-		widthCount =math.ceil(bunkerWidth/vehicle.cp.workWidth)
+		widthCount =math.ceil(bunkerWidth/(vehicle.cp.workWidth*0.7))--force overlape for compacketing and shovel driver
```
Comment (pvaiko) [inline BunkersiloManager.lua:29]: typo

**Comment by Tkaag** [comment]
I have tested the reverse unload and the changes I have made are not doing anything anymore. Is that can be dropped :)

File: CombineUnloadAIDriver.lua (hunk 1 above)
Comment (Tkaag) [reply CombineUnloadAIDriver.lua]: thangs for the update

**Comment by pvaiko** [comment]
So, what is the status on this?

**Pull request closed without merging**